### DISCUSSION TITLE
Reccommended approaches for actions, selectors, and tracked state

### DISCUSSION THREAD

**Discussion by MatthewScholefield** [udecode/zustand-x#80] [General]
Hello! As a relative newcomer to `zustand` and `zustand-x`, I feel a little lost about the "recommended" approach to build a real world app using `zustand-x`. Specifically, I'm wondering:

 1. If I can set state from anywhere, is it best to just define all single-use setters directly within components? Or is it still a good practice to define them directly on the store.
 2. We can define actions via either `set.state(draft => { draft.foo = 'bar'; draft.baz = 'boo'; })` or `set.foo('bar'); set.baz('boo')`. Is there a difference between the two (in terms of atomicity or performance)? I tend towards using the former since it supports easy nested assignment like `draft.foo[0].fooz = 'bar'`. Is it an anti-pattern to just have all actions using `set.state(draft => ...)`?
 3. When should we use tracked state versus using selectors? With pure `zustand` we could easily create on-the-fly selectors like `const isEmpty = useMyStore(x => x.myArr.length === 0)`. Is there a simple way to do this with `zustand-x`? (ie. is the suggested equivalent `myStore.useTracked.myArr.length === 0`?)
 
 Regardless, I'm finding `zustand-x` very useful even just for the typing aspect alone, but I'd appreciate if anyone could offer their insights or opinions on these ambiguities mentioned above. I think it'd be awesome to have a tiny, yet complete toy example that shows off the suggested way to design an app using `zustand-x`.

**Comment by zbeyens** [comment]
1. Up to you, I prefer ZustandX's DX
2. Use the former for batch updates, the latter otherwise (DX again).
3. You have access to the same store using `repoStore.useStore`

Here a few examples we have in Plate: https://github.com/search?q=repo%3Audecode%2Fplate%20createZustandStore&type=code
